lib/atecc508a/request.ex

```diff
@@ -196,6 +195,46 @@ defmodule ATECC508A.Request do
   defp length_flag(32), do: 1
   defp length_flag(4), do: 0
 
+  @spec transport_request(
+          transport :: Transport.t(),
+          payload :: binary(),
+          timeout :: non_neg_integer(),
+          response_payload_len :: non_neg_integer(),
+          request_timeout :: non_neg_integer()
+        ) :: {:ok, binary()} | {:error, atom()}
+  defp transport_request(transport, payload, timeout, response_payload_len, retry \\ 1000) do
+    now = System.monotonic_time(:second)
+    retry_request_with_timeout(transport, payload, timeout, response_payload_len, now, retry)
+  end
+
+  defp retry_request_with_timeout(transport, payload, timeout, response_payload_len, start, retry) do
+    result =
+      transport
+      |> Transport.request(payload, timeout, response_payload_len)
+      |> interpret_result()
+
+    case result do
+      {:ok, _} ->
+        result
+
+      {:error, :watchdog_about_to_expire} ->
+        if System.monotonic_time(:second) - start >= timeout,
+          do: {:error, :watchdog_retry_timeout},
+          else:
+            retry_request_with_timeout(
+              transport,
+              payload,
+              timeout,
+              response_payload_len,
+              start,
+              retry
+            )
```

Comment:
```suggestion
            retry_request_with_timeout(transport, payload, give_up_time, response_payload_len)
```

Reply:
I also wonder if we should `Process.sleep(100)` since if the ATECC is about to time out, it seems like we should wait a bit for it to timeout rather than asking again immediately. WDYT?